modules/programs/ssh.nix

```diff
@@ -282,7 +291,15 @@ let
   });
 
   matchBlockStr = cf: concatStringsSep "\n" (
-    ["Host ${cf.host}"]
+    (
+      if cf.exec != null
+      then [ (concatStringsSep " " (
+          ["Match"]
+          ++ optional (cf.host != null)          "host \"${cf.host}\""
+          ++ optional (cf.exec != null)          "exec \"${cf.exec}\""
+      )) ]
+      else ["Host ${cf.host}"]
```

Comment:
Can this be written as a `Match` so that we can unify the cases?

Reply:
I think so.
My main motivation here was backwards compability and familiarity with the generated config files.
I think that `Host foo bar` and `Match host "foo bar" ` should act the same, but I'll double-check to be sure.

Reply:
As I noted [here](https://github.com/nix-community/home-manager/pull/2370#issuecomment-934804666), this can't be rewritten without resulting in different semantics.